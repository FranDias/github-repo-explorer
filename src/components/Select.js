import React, { Component } from "react";

class App extends Component {

    render (
        <select onChange={this.selectSortCount} name="selectSortOption">
        {sortOptions.map(choice => (
          <option key={choice} value={choice}>
            Sort by {choice}
          </option>
        ))}
      </select>
    )
}

export default Select;