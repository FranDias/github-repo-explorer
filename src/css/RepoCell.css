.repo-cell {
  padding: 1em;
  flex-grow: 1
}

.repo-cell:first-of-type {
  width: 9em;
}

.repo-cell .label {
  opacity: 0.1;
  transition: opacity 300ms ease;
}

.repo-row:hover .repo-cell .label {
  opacity: 1
}

