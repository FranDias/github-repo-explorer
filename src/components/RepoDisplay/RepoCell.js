import React from "react";
import PropTypes from "prop-types";
import "./../../css/RepoCell.css";

const RepoCell = ({ label, url, text, expands }) => {
  const renderText = () => {
    if (!text) {
      return null;
    }
    let textContent = text;
    if (url) {
      textContent = <a href={url}>{textContent}</a>;
    }
    return <div>{textContent}</div>;
  };

  const getClassNames = () => {
    const expandClass = expands ? 'flex-grow-1' : '';
    return `repo-cell ${expandClass}`
  }

  return (
    <div className={getClassNames()}>
      <div className="label">{label}</div>
      {renderText()}
    </div>
  );
};

RepoCell.propTypes = {
  label: PropTypes.string,
  url: PropTypes.string,
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  expands: PropTypes.bool
};

RepoCell.defaultProps = {
  epxands: false,
  text: "none"
};

export default RepoCell;
