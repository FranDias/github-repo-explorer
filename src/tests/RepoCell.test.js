import React from "react";
import Select from "../components/Select";
import renderer from "react-test-renderer";
import RepoCell from "../components/RepoDisplay/RepoCell";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const repoCellProps = {
  label: "hi",
  text: "How is everyone?",
  expands: false
};
const repoCellPropsURL = {
  ...repoCellProps,
  url: "http://frandi.as",
  expands: true
};

describe("RepoCell", () => {
  describe("snapshots", () => {
    it("matches its snapshot with no props", () => {
      const component = renderer.create(<RepoCell />);
      expect(component).toMatchSnapshot();
    });

    it("matches snapshot with props", () => {
      const component = renderer.create(<RepoCell {...{ repoCellProps }} />);
      expect(component).toMatchSnapshot();
    });

    it("matches snapshot with props", () => {
      const component = renderer.create(<RepoCell {...{ repoCellPropsURL }} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe("renders props", () => {
    Object.keys(repoCellPropsURL).map(key => {
      it(`renders ${key}`, () => {
        const component = mount(<RepoCell {...repoCellPropsURL} />);
        expect(component.find(RepoCell).props()[key]).toEqual(
          repoCellPropsURL[key]
        );
      });
    });
  });
});
