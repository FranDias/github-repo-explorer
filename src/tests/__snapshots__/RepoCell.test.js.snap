// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoCell snapshots matches its snapshot with no props 1`] = `
<div
  className="repo-cell "
>
  <div
    className="label"
  />
  <div>
    none
  </div>
</div>
`;

exports[`RepoCell snapshots matches snapshot with props 1`] = `
<div
  className="repo-cell "
>
  <div
    className="label"
  />
  <div>
    none
  </div>
</div>
`;

exports[`RepoCell snapshots matches snapshot with props 2`] = `
<div
  className="repo-cell "
>
  <div
    className="label"
  />
  <div>
    none
  </div>
</div>
`;
